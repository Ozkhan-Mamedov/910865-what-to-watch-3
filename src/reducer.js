import {ALL_GENRES, CHANGE_FILTER_BY_GENRE, GET_FILMS_BY_GENRE} from "./constants";
import films from "./mocks/films";
import extend from "./utils";

const initialState = {
  genre: ALL_GENRES,
  films,
};

const ActionType = {
  CHANGE_FILTER_BY_GENRE,
  GET_FILMS_BY_GENRE,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_FILTER_BY_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case GET_FILMS_BY_GENRE:
      return extend(state, {
        films: films.filter((it) => it.genre === state.genre),
      });
  }

  return state;
};

export {reducer, ActionType};
