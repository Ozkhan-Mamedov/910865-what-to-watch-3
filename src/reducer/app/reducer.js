import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType} from "./action-creator";

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_FILTER_BY_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case ActionType.CHANGE_ACTIVE_CARD:
      return extend(state, {
        activeCard: action.payload,
      });

    case ActionType.INCREASE_CARDS_NUMBER:
      return extend(state, {
        cardsRenderNumber: state.cardsRenderNumber + action.payload,
      });

    case ActionType.DECREASE_CARDS_NUMBER:
      return extend(state, {
        cardsRenderNumber: action.payload,
      });

    case ActionType.CHANGE_SERVER_STATUS:
      return extend(state, {
        isServerAvailable: action.payload,
      });
  }

  return state;
};

export {reducer};
