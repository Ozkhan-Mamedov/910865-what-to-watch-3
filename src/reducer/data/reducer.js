import {commentAdapter, filmAdapter} from "../../adapter";
import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType, ActionCreator} from "./action-creator";

const Operation = {
  getFilmList: () => (dispatch, getState, api) => {
    return api.get(`/films`)
      .then((responce) => {
        dispatch(ActionCreator.getFilmList(
            responce.data.map((film) => filmAdapter(film))
        ));
      });
  },
  getCommentsList: (id) => (dispatch, getState, api) => {
    return api.get(`/comments/${id + 1}`)
      .then((responce) => {
        dispatch(ActionCreator.getCommentsList(
            responce.data.map((comment) => commentAdapter(comment))
        ));
      });
  },
  getPromoMovieData: () => (dispatch, getState, api) => {
    return api.get(`/films/promo`)
      .then((responce) => {
        dispatch(ActionCreator.getPromoMovieData(filmAdapter(responce.data)));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.GET_FILM_LIST:
      return extend(state, {
        films: action.payload,
      });

    case ActionType.GET_COMMENTS_LIST:
      return extend(state, {
        filmsComments: action.payload,
      });

    case ActionType.GET_PROMO_MOVIE_DATA:
      return extend(state, {
        promoFilm: action.payload,
      });
  }

  return state;
};

export {reducer, Operation};
