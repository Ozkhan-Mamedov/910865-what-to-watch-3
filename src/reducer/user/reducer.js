import extend from "../../utils";
import initialState from "./initial-state";
import {ActionType, ActionCreator} from "./action-creator";
import {AUTHORIZATION_STATUS} from "../../constants";

const Operation = {
  checkUserStatus: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.changeAuthorizationStatus(AUTHORIZATION_STATUS.AUTH));
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then(() => {
        dispatch(ActionCreator.changeAuthorizationStatus(AUTHORIZATION_STATUS.AUTH));
      });
  }
};

const reducer = (state = initialState, action) => {
  if (action.type === ActionType.CHANGE_AUTHORIZATION_STATUS) {
    return extend(state, {
      authorizationStatus: action.payload,
    });
  }

  return state;
};

export {reducer, Operation};
