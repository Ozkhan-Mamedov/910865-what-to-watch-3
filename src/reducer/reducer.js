import {combineReducers} from "redux";

import {reducer as data} from "./data/reducer";
import {reducer as app} from "./app/reducer";
import {reducer as user} from "./user/reducer";
import NameSpace from "./name-space";

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.APP]: app,
  [NameSpace.USER]: user,
});
