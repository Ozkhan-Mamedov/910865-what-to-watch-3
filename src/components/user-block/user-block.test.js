import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {createStore} from "redux";

import UserBlock from "./user-block";

import reducer from "../../reducer/reducer";

it(`UserBlock component renders correctly for main page`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(reducer)}>
          <UserBlock loginButtonClickHandler={() => {}}/>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
