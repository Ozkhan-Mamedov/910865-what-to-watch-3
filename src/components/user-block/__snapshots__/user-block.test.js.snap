// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component renders correctly for main page 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="sign-in.html"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</div>
`;
